api/v1beta1/azuremanagedcontrolplanetemplate_default.go

```diff
@@ -37,8 +37,8 @@ func (mcp *AzureManagedControlPlaneTemplate) setDefaults() {
 
 	mcp.setDefaultVirtualNetwork()
 	mcp.setDefaultSubnet()
-	mcp.Spec.Template.Spec.SKU = setDefaultSku(mcp.Spec.Template.Spec.SKU)
-	mcp.Spec.Template.Spec.AutoScalerProfile = setDefaultAutoScalerProfile(mcp.Spec.Template.Spec.AutoScalerProfile)
+	setDefaultSku(mcp.Spec.Template.Spec.SKU)
+	setDefaultAutoScalerProfile(mcp.Spec.Template.Spec.AutoScalerProfile)
```

Comment:
I have no self-control and can't stop playing with generics, so offering this in case it's useful:
```go
func setDefault[T comparable](field *T, value T) {
	if field == nil {
		// shouldn't happen with proper use
		return
	}
	var zero T
	if *field == zero {
		*field = value
	}
}

setDefault(&m.Spec.NetworkPlugin, ptr.To(AzureNetworkPluginName))
//         ^ where referencing a field (even if it's already a pointer) is "proper use"

func TestSetDefault(t *testing.T) {
	g := NewGomegaWithT(t)

	type Struct struct{ name string }

	var s *Struct
	setDefault(&s, &Struct{"hello"})
	g.Expect(s.name).To(Equal("hello"))
	setDefault(&s, &Struct{"world"})
	g.Expect(s.name).To(Equal("hello"))

	r := &Struct{}
	setDefault(&r, &Struct{"a name"})
	g.Expect(r.name).To(BeEmpty())
	setDefault(&r.name, "hello")
	g.Expect(r.name).To(Equal("hello"))
	setDefault(&r.name, "world")
	g.Expect(r.name).To(Equal("hello"))

	str := ""
	setDefault(&str, "a string")
	g.Expect(str).To(Equal("a string"))
	setDefault(&str, "another string")
	g.Expect(str).To(Equal("a string"))
}


```

Reply:
Nice! I incorporated this into as many checks in `Default()` as I could. Some checks that didn't perform a nil check couldn't use this.